### DISCUSSION TITLE
Multiple Keys Per Item

### DISCUSSION THREAD

**Discussion by dstark** [retorquere/zotero-better-bibtex#3032] [General]
Is it possible to have multiple citation keys for a given item? For instance, if some citations have been created on one key, is there a way of saving that key so that it will still be usable by those citations so that they won't break if a different key is subsequently assigned to that item (e.g., due to a change in preference for how the citation key formula is configured)?

**Comment by retorquere** [comment]
Yes; you can add a `tex.ids: <other keys>` to the `extra` field. BibTeX will pick those up.

**Reply by dstark** [reply]
> Yes; you can add a `tex.ids: <other keys>` to the `extra` field. BibTeX will pick those up.

Thanks so much. I see that `tex.ids: <other keys>` seems, though, not to be included in a Better CSL JSON export. Should I be listing additional keys differently if that's the export I'm intending to generate?

**Reply by retorquere** [reply]
CSL doesn't have standard fields for this, but you can do this in a BBT postscript:

```
if (Translator.BetterCSL) {
  if (zotero.extraFields && zotero.extraFields.aliases) {
    csl.citationkeys = zotero.extraFields.aliases
  }
}
```

the BBT prefs screen is broken though currently, I'm working to bring it back up.
